### DISCUSSION TITLE
Automatic update of location with GPS

### DISCUSSION THREAD

**Discussion by Kytutr** [mcguirepr89/BirdNET-Pi#1189] [Ideas]
Imagine a situation when you're moving around with your BirdNET-Pi running on Raspberry Pi. You may change your location many times a day. Now you must change location in settings every time you move to a different spot (to keep consistency of your recordings). 

With my scripts, a location will be discovered and updated automatically. All it takes is to connect GPS device to Raspberry Pi (or different device of your choice), ensure that gpsd service is running properly, enable location autoupdate and then you can forget it.

I already have the pull request for that, but it's for Nachtzuster branch, as I'm active user BirdNET-Pi flavor:
https://github.com/Nachtzuster/BirdNET-Pi/pull/42 

And it's also ready to pick up and test from my branch: https://github.com/Kytutr/BirdNET-Pi (but still PR above is much easier to grasp all changes).

How it looks in my system:
![image](https://github.com/mcguirepr89/BirdNET-Pi/assets/33352597/31a68e45-243b-491a-8ead-4176ad333b0b)

![image](https://github.com/mcguirepr89/BirdNET-Pi/assets/33352597/da173d76-5fb9-439a-ba53-8fe19e5b9b3c)

![image](https://github.com/mcguirepr89/BirdNET-Pi/assets/33352597/7f4691e9-199f-4306-bbed-863c29b6b180)

What do you think about it?

**Comment by Inachis** [comment]
This is timely as I just got a NEO-7M GPS module connected to a Pi and working with gpsd prior to looking at integration with BirdNET-Pi!

I won't get a chance to try your update until the weekend. Does it set the Pi clock to GPS time? 

Kelvin

**Reply by Kytutr** [reply]
I didn't know that I have that great timing :-) 
And fetching time from GPS is a great idea - such a simple and useful thing. I will try to code it over this weekend. Thank you for sharing this thought @Inachis

**Comment by Svardsten53** [comment]
It looks really good. 

I can see a problem. If you upload data to BirdWeather, it will probably fail when you change position, because the position in BirdNET-Pi must be the same as the one reported to BirdWeather.

**Reply by Kytutr** [reply]
Can you explain to me how BirdWeather works? Do you set the position in BirdWeather? 

To mitigate that problem, I can do two things:
- put a warning on settings page to not enable location autoupdate when using BirdWeather (the easy and lazy way)
- automatically disable location update when BirdWeater is configured (more coding involved)

Thanki you for pointing this problem and for nice words about my idea, @Svardsten53

**Comment by Svardsten53** [comment]
In BirdNET-Pi settings you can apply for a BirdWeather ID by sending an email where you specify information about name, title and latitude and longitude which will be automatically fetched from your configuration. All detections sent to BirdWeather must match the location you have sent in. I'm not familiar how BirdWeather internally works. 

Unfortunately, this version of BirdNET-Pi is no longer being updated, but your code can certainly be added to the fork that is currently being updated. https://github.com/Nachtzuster/BirdNET-Pi

**Comment by Inachis** [comment]
It took me too long to get to integrating. I have the branch from https://github.com/Kytutr/BirdNET-Pi installed and straight away identifying birds in the garden. But I don't know how to configure gpsd for the experimental support of mobile location recording. Will it set the time from GPS too?
